app/controllers/application_controller.rb

```diff
@@ -1,10 +1,156 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    use Rack::Flash
+    set :session_secret, "ek_and_mk_for_lyfe"
+  end
+
+  helpers do
+    def current_user
+      User.find(session[:user_id])
+    end
+
+    def is_logged_in?(session)
+      session.has_key?(:user_id)
+    end
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/tweets' do
+    if is_logged_in?(session)
+      @tweets = Tweet.all.sort_by {|tweet| tweet.id}.reverse
+      erb :'/tweets/tweets'
+    else
+      flash[:error] = "Please log in to view tweets."
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if is_logged_in?(session)
+      erb :'/tweets/create_tweet'
+    else
+      flash[:error] = "Please log in or create an account to start tweeting."
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] != ""
+      Tweet.create(content: params[:content], user_id: session[:user_id])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.